openvdb/python/CMakeLists.txt

```diff
@@ -79,16 +79,17 @@ IF ( USE_BLOSC )
   FIND_PACKAGE ( Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED )
 ENDIF ()
 
+# NumPy requires CMake 3.14
+IF ( USE_NUMPY AND ( ${CMAKE_VERSION} VERSION_LESS 3.14 ) )
+  MESSAGE ( FATAL_ERROR "No CMake support for FindPackage ( NumPy ) currently "
+    "available for OpenVDB. Requires CMake 3.14." )
```

Comment:
Are you sure about this? CMake 3.14.0 was only released two days ago:

https://github.com/Kitware/CMake/releases/tag/v3.14.0

In comparison, NumPy 1.14.5 (VFX Reference Platform 2019 states NumPy 1.14.x) was released in June 2018:

https://github.com/numpy/numpy/releases/tag/v1.14.5

Reply:
Yep I double checked, I thought it was 3.12 but that was when the FindPython modules were added with Dev, Interpreter and Compiler support. NumPy was only just added. See https://cmake.org/cmake/help/latest/release/3.14.html

Reply:
I agree we should talk about which CMake minimum version we should support. Either way, I think we'd need NumPy support in a CMake version older than 3.14. 

Just as a starting point for discussion, if we were to use a similar lifetime/deprecation strategy for CMake as we do the VFX Reference Platform years - the earliest CMake release in 2017 was 3.7.2, so we should support building against NumPy for 3.7.2 until now. That could perhaps mean backporting the FindPackage for NumPy into our codebase until we reach a minimum of 3.14.

Reply:
That make sense to me. All other packages which support numpy incorporate their own find modules. #160 also does this, mentioned in the overview above. The module seems to have been copied from elsewhere, so we could either use that or just copy the NumPy branch in CMakes FindPython/Support.cmake stuff.

Reply:
I see nothing wrong with custom `Find` files for modules that don't yet have built-in support.  It's unreasonable to expect users to jump to 3.14.

Reply:
As discussed in TSC, will hold off from adding more numpy support in this PR. There are also issues where Boost > 1.65 which require boost numpy which I'll resolve later as well. Added to PR description of known issues.